### PR TITLE
Feature/integration tests

### PR DESCRIPTION
**Pull request by kzalite**

### CODE REVIEW

File: test/endToEnd/util/createSubproblem.json [hunk 1]
```diff
@@ -0,0 +1,222 @@
+{
```
Comment (DanielReid) [inline test/endToEnd/util/createSubproblem.json]: filename seems weird, this reads like a function name

File: test/endToEnd/util/setPvf.json [hunk 2]
```diff
@@ -0,0 +1,94 @@
+{
```
Comment (DanielReid) [inline test/endToEnd/util/setPvf.json]: filename reads like function name

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+name: MCDA CI
+
+on:
+  push:
+    branches: feature/nightwatch
```
[the remark this answers is not in the document]
Comment (chinook25) [reply .github/workflows/main.yml]: Yes, this file should not be in this commit, removed it

File: package.json [hunk 4]
```diff
@@ -1,15 +1,17 @@
...
   "scripts": {
     "start": "node index",
     "test": "mocha && karma start --singleRun",
+    "test-end-to-end": "nightwatch -c test/endToEnd/nightwatch.conf.js",
     "build-dev": "webpack --config webpack.dev.js",
     "build-prod": "webpack --config webpack.prod.js",
     "build-local-login": "webpack --config webpack.local.js",
-    "dev": "webpack --config webpack.dev.js --watch"
+    "dev": "webpack --config webpack.dev.js --watch",
+    "dev-local": "webpack --config webpack.local.js --watch"
```
[the remark this answers is not in the document]
Comment (chinook25) [reply package.json]: made story for it on trello.

File: test/endToEnd/closeModalTest.js [hunk 5]
```diff
@@ -0,0 +1,224 @@
+'use strict';
+
+const loginService = require('./util/loginService');
+const workspaceService = require('./util/workspaceService');
+
+const testUrl = 'http://localhost:3002';
+
+const title = 'Antidepressants - single study B/R analysis (Tervonen et al, Stat Med, 2011)';
+const preferenceTabPath = '//*[@id="preferences-tab"]';
+const rankingCellPath = '//*[@id="de14e778-f723-48d4-8f4e-1e589714f4f2-ranking"]';
```
[the remark this answers is not in the document]
Comment (chinook25) [reply test/endToEnd/closeModalTest.js]: created trello card for this

File: test/endToEnd/closeModalWorkspacesTest.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const loginService = require('./util/loginService');
+const testUrl = 'http://localhost:3002';
+const closeModalButton = '//*[@id="close-modal-button"]';
+const deleteWorkspaceButton = '//div[2]/div/div[2]/workspaces/div/div[2]/table/tbody/tr/td[3]/a';
+
+module.exports = {
+  beforeEach: function(browser) {
+    browser.resizeWindow(1366, 728);
```
[the remark this answers is not in the document]
Comment (chinook25) [reply test/endToEnd/closeModalWorkspacesTest.js]: created trello card for this

File: test/endToEnd/scenarioTest.js [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+  afterEach: function(browser) {
+    browser.click('#logo');
+    workspaceService.deleteFromList(browser, title);
+    browser.end();
+  },
+
+  'Creating a new scenario': function(browser) {
+    browser
+      .click('#create-scenario-button')
+      .waitForElementVisible('#create-new-scenario-button:disabled')
```
[the remark this answers is not in the document]
Comment (chinook25) [reply test/endToEnd/scenarioTest.js]: I don't really understand, this is about 2 different buttons, we want to click the first and check if the second is disabled when the modal opens.

File: test/endToEnd/util/util.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    const elementId = getOnlyProperty(result.value);
+    browser.elementIdDisplayed(elementId, function(isDisplayedResult) {
+      chai.expect(isDisplayedResult.value).to.be.false;
+    });
+  });
+}
+
+function isElementNotPresent(browser, path) {
+  browser.element('xpath', path, function(result) {
+    // this test outputs an error message to the console but does not make the overall test fail
```
[the remark this answers is not in the document]
Comment (chinook25) [reply test/endToEnd/util/util.js]: nope

File: test/endToEnd/subproblemTest.js [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+}
+
+module.exports = {
+  beforeEach: function(browser) {
+    loginService.login(browser, testUrl, loginService.username, loginService.correctPassword);
+    workspaceService.uploadTestWorkspace(browser, workspacePath);
+  },
+
+  afterEach: function(browser) {
+    errorService.isErrorBarHidden(browser);
```
[the remark this answers is not in the document]
Comment (chinook25) [reply test/endToEnd/subproblemTest.js]: We wanted to reduce the log spam, that's why we don't check it everywhere.

File: test/endToEnd/manualInputTest.js [hunk 10]
```diff
@@ -0,0 +1,383 @@
...
+
+  'Editing a criterion': function(browser) {
+    const newTitle = 'newTitle';
+    const newDescription = 'newDescription';
+
+    createInputDefault(browser);
+    browser
+      .useXpath()
+      .click(criterion1EditPath)
+      .useCss()
```
[the remark this answers is not in the document]
Comment (chinook25) [reply test/endToEnd/manualInputTest.js]: We think the css selectors are more readable, could include changing this in a refactor.

File: app/js/effectsTable/criterionCardDirective.html [hunk 11]
```diff
@@ -1,15 +1,15 @@
 <div class="grid-x callout">
-  <div ng-if="editMode.isUserOwner" class="cell shrink" style="display:flex; align-items:center;">
+  <div ng-if="isStandalone" class="cell shrink" style="display:flex; align-items:center;">
```
[the remark this answers is not in the document]
Comment (kzalite) [reply app/js/effectsTable/criterionCardDirective.html]: Rolled back, not relevant to this story, and also incorrect.

File: test/endToEnd/copyWorkspaceTest.js [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+    const title = 'Antidepressants - single study B/R analysis (Tervonen et al, Stat Med, 2011)';
+    const newTitle = 'copy of a workspace';
+    
+    loginService.login(browser, testUrl, loginService.username, loginService.correctPassword);
+    workspaceService.addExample(browser, title);
+    workspaceService.copy(browser, title, newTitle);
+    workspaceService.deleteFromList(browser, newTitle);
+    workspaceService.deleteFromList(browser, title);
+      
+    browser.end();
```
[the remark this answers is not in the document]
Comment (kzalite) [reply test/endToEnd/copyWorkspaceTest.js]: We test deletion separately, no need to assert again?

File: test/endToEnd/scenarioTest.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (kzalite) [reply test/endToEnd/scenarioTest.js]: The current solution for checking whether the button is disabled is the easiest way to implement it. Here, the test is about the 'disabled' property and not that the button is showing at all.

File: test/endToEnd/subproblemTest.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (DanielReid) [reply test/endToEnd/subproblemTest.js]: with the fix this can now be used everywhere :)

**Pull request merged**